app/src/main/java/com/github/sdp_begreen/begreen/firebase/FirebaseDB.kt

```diff
@@ -0,0 +1,162 @@
+
+    private val TAG: String = "Firebase Database"
+    private val TIMEOUT: Long = 10000 // Waiting time before aborting a <get> on the database
+    private val ONE_MEGABYTE: Long = 1024 * 1024 // Maximal image size allowed (in bytes), to prevent out of memory errors
+    // Realtime database ref
+    private val databaseReference: DatabaseReference = Firebase.database.reference
+    // Storage ref (for images)
+    private val storageReference: StorageReference = Firebase.storage.reference
+    private val connectedReference = Firebase.database.getReference(".info/connected")
+
+    // Logs (in the console) the connections and disconnections with the Firebase database
+    // We might want to provide a new constructor that takes code to execute on connections/disconnections
+    // (to show the user when he gets connected/disconnected by example)
+    init {
+        connectedReference.addValueEventListener(object : ValueEventListener {
+            override fun onDataChange(snapshot: DataSnapshot) {
+                if (snapshot.getValue(Boolean::class.java) == true) {
+                    Log.d(TAG, "connected")
+                } else {
+                    Log.d(TAG, "disconnected")
+                }
+            }
+
+            override fun onCancelled(error: DatabaseError) {
+                Log.d(TAG, "Connection listener was cancelled")
+            }
+        })
+    }
+
+    /**
+     * Returns the value associated to the to given [key] in the database
+     *
+     * @param key the key we want to know the value of
+     * @return the value associated to the [key]
+     * @throws DatabaseException if the value cannot be retrieved
+     */
+    @SuppressLint("RestrictedApi")
+    suspend fun get(key: String): String? {
+
+        return try {
+            // Cancel the query and throw exception after TIMEOUT ms
+            withTimeout(TIMEOUT) {
+                val data = databaseReference.child(key).get().await()
+                data.value?.let { it as? String }
+            }
+        } catch (timeOutEx: TimeoutCancellationException) {
+            Log.d(TAG, "Timeout, cant connect with database")
+            throw DatabaseException("Timeout, cant connect with database")
+        }
+        catch (databaseEx: DatabaseException) {
+            Log.d(TAG, "Failed with error message: ${databaseEx.message}")
+            throw databaseEx
+        }
+    }
+
+    /**
+     * Associate a new [value] to the given [key].
+     * Create a new entry if the [key] did not exist
+     *
+     * @param key the key we want to set the value (non empty)
+     * @param value the new value for the [key]
+     */
+    operator fun set(key: String, value: String) {
+        if (key.isNotEmpty()) databaseReference.child(key).setValue(value)
+    }
+
+    /**
+     * Store the given [image] under the ID [userId]
+     * and returns the imageId set for that image in the database
+     * Returns null if the image can't be stored
+     */

```

Comment:
This comment looks orphaned, as if it was applying to a method that no longer exists

Reply:
That's probably the case. David or me must have spotted in when working on `FirebaseDB` in our later tasks, because it is not present anymore. Thanks though!